src/components/DeviceInteraction/DeviceInteractionStep.js

```diff
@@ -0,0 +1,225 @@
+// @flow
+
+import React, { PureComponent } from 'react'
+
+import Box from 'components/base/Box'
+import { delay } from 'helpers/promise'
+
+import {
+  DeviceInteractionStepContainer,
+  SpinnerContainer,
+  IconContainer,
+  SuccessContainer,
+  ErrorDescContainer,
+  ErrorContainer,
+} from './components'
+
+export type Step = {
+  id: string,
+  title: string | (any => string),
+  desc: any,
+  icon: React$Element<*> | null,
+  run?: any => Promise<any> | { promise: Promise<any>, unsubscribe: void => any },
+  render?: (
+    { onSuccess: any => any, onFail: any => any, onRetry: any => any },
+    any,
```

Comment:
it would be better to try to not have to use `any` , so we have more flow checking